src/librustc_codegen_ssa/back/write.rs

```diff
@@ -147,6 +148,8 @@ impl ModuleConfig {
             || sess.opts.cg.linker_plugin_lto.enabled()
         {
             EmitObj::Bitcode
+        } else if sess.target.target.options.forces_embed_bitcode {
```

Comment:
Could this get folded into `need_crate_bitcode_for_rlib`?

Reply:
This is interesting. Right now I can't see any code path for embedding `Marker` bitcode. Do we need it? IMO probably not. Would it make any sense for `OptLevel` to downgrade the wishes of my target spec from `Full` to `Marker`? Again probably not. It's certainly surprising, and it complicates the situation for someone debugging Rust in Xcode.

So at the risk of being over-enthusiastic, I've codified this by removing the `Marker` variant along with the ineffectual `OptLevel` match. Do you reckon this makes sense?

Reply:
Oh thanks for the keen eye! I think that may have actually been an issue from an earlier one of my PRs, but at this point I think it's safe to remove. We can always add it back in later if truly necessary.